pallets/collator-allowlist/Cargo.toml

```diff
@@ -14,6 +14,7 @@ targets = ['x86_64-unknown-linux-gnu']
 codec = { package = 'parity-scale-codec', version = '2.0.0', features = ['derive'] , default-features = false }
 scale-info = { version = "1.0", default-features = false, features = ["derive"] }
 sp-std = { git = "https://github.com/paritytech/substrate", default-features = false , branch = "polkadot-v0.9.12" }
+frame-benchmarking = { git = "https://github.com/paritytech/substrate",  default-features = false , optional = true , branch = "polkadot-v0.9.12" }
```

Comment:
lets move all the optional deps seperately.

Reply:
That doesn't seem possible unfortunately. 
If I move this to `[dev-dependencies]`, I need to drop the `optional = true` flag, or else it fails with:

```
dev-dependencies are not allowed to be optional: `frame-benchmarking`
```

But by keeping it, we get:

```
feature `runtime-benchmarks` includes `frame-benchmarking`, but `frame-benchmarking` is not an optional dependency
A non-optional dependency of the same name is defined; consider adding `optional = true` to its definition.
```

Reply:
I meant moving to different group within dependcies so its easy to understand what are hard deps and what are benchmarking deps. Something like this - https://github.com/centrifuge/centrifuge-chain/blob/tocc/pallets/loan/Cargo.toml#L24